nvflare/fuel/message/event_manager.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Optional
+
+from nvflare.fuel.message.data_bus import DataBus
+
+
+class EventManager:
+    """
+    Class for managing events by interacting with a DataBus.
+
+    Args:
+        data_bus (DataBus): An instance of the DataBus class used for event communication.
+    """
+
+    def __init__(self, data_bus: "DataBus"):
+        """
+        Initialize the EventManager with a DataBus instance.
+
+        Args:
+            data_bus (DataBus): An instance of the DataBus class used for event communication.
+        """
+        self.data_bus = data_bus
+
+    def fire_event(self, event_name: str, event_data: Optional[Any] = None) -> None:
+        """
```

Comment:
I guess I know what you are trying to do now: the DataBus is a storage mechanism for data, and at the same time you want to be able to fire events. The "data" here is not the same as the data saved into the bus.

These two needs are independent with each other. DataBus should first be a data store/sharing mechanism for keyed data. Secondly, you could define a Pub/Sub interface and make DataBus also implement this interface (by the same token, we could make FLContext also implement the Pub/Sub interface).

The callback signature for pub/sub should include: topic, the event data, and the databus itself since the CB will need to access the data bus for stored data.